src/checks/truncref.py

```diff
@@ -0,0 +1,195 @@
+import sys
+import re
+import dns.name
+import dns.message
+import dns.query
+import dns.exception
+
+
+
+#Loop that tries all servers in case some of them are down/unavailable
+def __ask_servers(list_of_servers, request):
+    counter = 0
+    for server in list_of_servers:
+        try:
+            response = dns.query.udp(request, server, 0.3)          #Send a request to a DNS server and get the reply
+            return ("OK", response)
+        except dns.exception.Timeout:                               #If there's a timeout the loop continues and tries the next server
+            counter += 1
+            if counter == list_of_servers.__len__():                #If all servers are unavailable then the test fails
+                return ("Failed", None)
+            else:
+                continue
+
+
+#Checks for a server IP addresses from the given section of a DNS server's response 
+def __parse_records(list_of_records, pattern, group):
+ counter = 0
+ servers = []
+
+ #Check all records of response section
+ for text_response in list_of_records:
+    counter += 1
+
+    record = re.search(pattern, text_response.to_text(), re.VERBOSE)         #Match it with the given regex pattern
+    
+    if record != None:                                                       #If the match succeeds add the selected part of the record to a list
+        servers.append(record.group(group))
+
+    if counter == list_of_records.__len__() and servers.__len__() == 0:     #If there are no matches with the regex then the parsing fails
+        return ("Failed", [])
+
+    if  counter == list_of_records.__len__():                               #When all records have been checked, exit loop and return a list of the selected fields
+        return ("OK", servers)
+
+
+
+
+def run(domain, list_of_servers):
+
+    details = []
+    #Goes through the list of authoritative name servers for the domain and checks each one
+    for server in list_of_servers:
+
+        result = __truncref(domain, server)
+        details.append(result)
+        if not result.get('result'):
+            return {"description": "Check for {0} failed with: {1}".format(server, result.get('description')), "result":False, 'details': details}             #If one of them fails the check stops immediately 
+        
+        else:
+            continue
+    
+    return {"description": "All authoritative name servers passed!", "result":True, "details": details}                                                         #If loop exits without returning all servers passed
+
```

Comment:
run should have consistent return descriptions

Reply:
They already have consistent descriptions.

Reply:
> run should have consistent return descriptions

They already have consistent descriptions

Reply:
No.
Description is the label of the check which is shown to the user.

If user sees:
`FAIL - Check for ns1.loopia.se failed with: xxxx`
This will not tell the user what check exactly failed.
Or
`PASS - All authoritative name servers passed!`
How does user know what it passed?

It could show something like this:
FAIL - No truncation of referrals

Reply:
or, maybe like this since the "no" word is confusing:
FAIL - truncation of referrals
PASS - No truncation of referrals

Reply:
The name "Truncation of referrals" is only used in the IANA page and doesn't really explain the check, IMO.